# HG changeset patch
# User Rik <rik@octave.org>
# Date 1419957111 28800
#      Tue Dec 30 08:31:51 2014 -0800
# Node ID 8ee14c64ab5fca22acf99890aafb9c2873c46d11
# Parent  93fbdbcb3349fa476939909beea3b8eeed565cfb
Document Matlab incompatibility in printf functions and bare '%' char (bug #42345).

* io.txi: Document Matlab incompatibility in printf functions and bare '%'
char.

diff --git a/doc/interpreter/io.txi b/doc/interpreter/io.txi
--- a/doc/interpreter/io.txi
+++ b/doc/interpreter/io.txi
@@ -611,20 +611,33 @@ Print a single character.  @xref{Other O
 @item @samp{%s}
 Print a string.  @xref{Other Output Conversions}.
 
 @item @samp{%%}
 Print a literal @samp{%} character.  @xref{Other Output Conversions}.
 @end table
 
 If the syntax of a conversion specification is invalid, unpredictable
-things will happen, so don't do this.  If there aren't enough function
-arguments provided to supply values for all the conversion
-specifications in the template string, or if the arguments are not of
-the correct types, the results are unpredictable.  If you supply more
+things will happen, so don't do this.  In particular, @sc{matlab} allows
+a bare percentage sign @samp{%} with no subsequent conversion character.
+Octave will emit an error and stop if it sees such code.  When the string
+variable to be processed cannot be guaranteed to be free of potential format
+codes it is better to use the two argument form of any of the @code{printf}
+functions and set the format string to @code{%s}.
+
+@example
+@group
+printf (strvar);        # Unsafe if strvar contains format codes
+printf ("%s", strvar);  # Safe
+@end group
+@end example
+
+If there aren't enough function arguments provided to supply values for all
+the conversion specifications in the template string, or if the arguments are
+not of the correct types, the results are unpredictable.  If you supply more
 arguments than conversion specifications, the extra argument values are
 simply ignored; this is sometimes useful.
 
 @node Integer Conversions
 @subsection Integer Conversions
 
 This section describes the options for the @samp{%d}, @samp{%i},
 @samp{%o}, @samp{%u}, @samp{%x}, and @samp{%X} conversion
